tools/cli/include/nearobject/cli/NearObjectCliData.hxx

```diff
@@ -93,6 +95,46 @@ struct UwbApplicationConfigurationParameterData
     std::optional<std::unordered_set<uwb::UwbMacAddress>> destinationMacAddresses;
     std::optional<uwb::protocol::fira::DeviceType> deviceType;
     std::optional<uwb::UwbMacAddressType> macAddressMode;
+
+    /**
+     * @brief Variant for all possible property types.
+     */
+    using ParameterTypesVariant = std::variant<
+        uint8_t,
+        ::uwb::protocol::fira::DeviceRole,
+        ::uwb::protocol::fira::DeviceType,
+        ::uwb::protocol::fira::MultiNodeMode,
+        ::uwb::UwbMacAddress,
+        ::uwb::UwbMacAddressType,
+        std::unordered_set<::uwb::UwbMacAddress>>;
+
+    /**
+     * @brief Map of application configuration parameter type to the parameter's value.
+     */
+    std::unordered_map<UwbApplicationConfigurationParameterType, ParameterTypesVariant>
+    GetValueMap() const
+    {
+        std::unordered_map<UwbApplicationConfigurationParameterType, ParameterTypesVariant> valuesMap;
+        if (deviceRole.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::DeviceRole] = deviceRole.value();
+        }
+        if (multiNodeMode.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::MultiNodeMode] = multiNodeMode.value();
+        }
+        if (numberOfControlees.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::NumberOfControlees] = numberOfControlees.value();
+        }
+        if (deviceMacAddress.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::DeviceMacAddress] = deviceMacAddress.value();
+        }
+        if (destinationMacAddresses.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::DestinationMacAddresses] = destinationMacAddresses.value();
+        }
+        if (deviceType.has_value()) {
+            valuesMap[UwbApplicationConfigurationParameterType::DeviceType] = deviceType.value();
+        }
+        return valuesMap;
+    }

```

Comment:
Can you make this a declaration and move the definition into `NearObjectCliData.cxx` please?

In general, please avoid putting code in headers. There are a whole bunch of good reasons for this: separating the implementation from the interface, ease of debugging, reducing build times, etc. The only cases I can think of where it's reasonable to have code in a header is for templates, `constexpr` or `consteval` expressions, and `default`, compiler-generated member functions; all of those are required by the standard to be in the header.

Reply:
Sure, I totally overlooked that this was in the header. Will move